packages/blade/src/components/Tooltip/__tests__/Tooltip.web.test.tsx

```diff
@@ -0,0 +1,250 @@
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
+/* eslint-disable @typescript-eslint/require-await */
+/* eslint-disable @typescript-eslint/no-empty-function */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { act, fireEvent, screen } from '@testing-library/react';
+import React from 'react';
+import type { BladeCommonEvents } from '..';
+import { Tooltip, TooltipInteractiveWrapper } from '..';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.web';
+import { Button } from '~components/Button';
+import { paymentTheme } from '~tokens/theme';
+import assertAccessible from '~src/_helpers/testing/assertAccessible.web';
+import { InfoIcon } from '~components/Icons';
+import { Text } from '~components/Typography';
+import BaseBox from '~components/Box/BaseBox';
+
+const waitForPosition = () => act(async () => {});
+const animationDuration = paymentTheme.motion.duration.quick;
+
+describe('<Tooltip />', () => {
+  jest.useFakeTimers();
+
+  it('should render', () => {
+    const buttonText = 'Hover me';
+    const { container, getByRole } = renderWithTheme(
+      <Tooltip content="Hello world">
+        <Button>{buttonText}</Button>
+      </Tooltip>,
+    );
+
+    // snapshot while on opened
+    fireEvent.focus(getByRole('button', { name: buttonText }));
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should open on hovering over', async () => {
+    const buttonText = 'Hover me';
+    const { getByRole } = renderWithTheme(
+      <Tooltip content="Hello world">
+        <Button>{buttonText}</Button>
+      </Tooltip>,
+    );
+    await waitForPosition();
+
+    fireEvent.mouseEnter(getByRole('button', { name: buttonText }));
+
+    // should not be opened by default
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+    // should open after 300ms
+    await act(async () => {
+      jest.advanceTimersByTime(299);
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    // close after 300ms
+    fireEvent.mouseLeave(getByRole('button', { name: buttonText }));
+    await act(async () => {
+      jest.advanceTimersByTime(299 + animationDuration);
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+  });
+
+  it('should open/close immediately on focus/blur without the default delay of 300ms', async () => {
```

Comment:
should just open immediately right? because in the test you're waiting for the animation to completely on blur so I'm assuming it'll close after 300ms?

Reply:
It will close and open immediately on focus and blur, keyboard users can't express "I'm going to hover and stay here for a few miliseconds" intent they instantly focus/blur to the elements.